src/json_parser.cc

```diff
@@ -58,23 +61,50 @@ bool JSONParser::Parse(const std::string& content) {
   return true;
 }
 
-std::optional<std::string> JSONParser::GetTopLevelField(
-    const std::string& field) {
+Maybe<std::string> JSONParser::GetTopLevelStringField(std::string_view field) {
```

Comment:
Why is this changed to `v8::Maybe` here? I'd prefer we leave the option of making `JSONParser` independent of V8 open (e.g. for parsing other configs that should be read before V8 is even initialized, like the NODE_OPTIONS in package.json idea).

Reply:
@joyeecheung This one is because of @addaleax's suggestion in https://github.com/nodejs/node/pull/47588#discussion_r1168752918. These functions were returning `std::optional`s but with the semantics of `v8::Maybe`. A `v8::Maybe` conveys the meaning that we shouldn't call into V8 if an exception is pending. Using `std::optional` doesn't really make that obvious.

> for parsing other configs that should be read before V8 is even initialized, like the NODE_OPTIONS in package.json idea

Do you mean regarding the implementation for embedding the prebaked CLI args in the SEA config? I think the JSONParser class would only be involved in the part where the user's SEA JSON config is read and the blob is constructed, where I think it should be okay to do this after V8 is initialized, i.e., what we are doing right now. When the SEA runs, the injected blob would be read in the `FindSingleExecutableResource` function which parses the buffer that `postject_find_resource` returns and constructs the `SeaResource` object without using the JSONParser class anywhere.

Reply:
I don’t think it’s a bad idea to keep the option of making `JSONParser` independent of V8, but then its API contract should also be explicit about this (and in particular not leak exceptions to callers).

Reply:
Fair point, I've reverted the optional->Maybe change